PROTO_tests/tests/series_test.py

```diff
@@ -0,0 +1,186 @@
+            mi = ak.MultiIndex([pda, pda[::-1]])
+            s = ak.Series(data=val, index=mi)
+            lk = s.locate(mi[0])
+            assert isinstance(lk, ak.Series)
+            assert lk.index.index == mi[0].index
+            assert lk.values[0] == val[0]
+
+            # ensure error with scalar and multi-index
+            with pytest.raises(TypeError):
+                s.locate(0)
+
+            with pytest.raises(TypeError):
+                s.locate([0, 2])
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NO_STRING)
+    def test_add(self, prob_size, dtype):
+        added = ak.Series(ak.arange(prob_size // 2, dtype=dtype)).add(
+            ak.Series(
+                data=ak.arange(prob_size // 2, prob_size, dtype=dtype),
+                index=ak.arange(prob_size // 2, prob_size),
+            )
+        )
+        assert (added.index == ak.arange(prob_size)).all()
+        assert (added.values == ak.arange(prob_size, dtype=dtype)).all()
+
+    @pytest.mark.parametrize("dtype", [ak.int64, ak.uint64, ak.float64])
+    def test_topn(self, dtype):
+        top = ak.Series(ak.arange(100, dtype=dtype)).topn(50)
+        assert top.values.to_list() == list(range(99, 49, -1))
+        assert top.index.to_list() == list(range(99, 49, -1))
+
+    @pytest.mark.parametrize("dtype", NUMERICAL_TYPES)
+    @pytest.mark.parametrize("dtype_index", NUMERICAL_TYPES)
+    def test_sort(self, dtype, dtype_index):
+        def gen_perm(n):
+            idx_left = set(range(n))
+            perm = np.arange(n)
+            while len(idx_left) >= 2:
+                inds = np.random.choice(list(idx_left), 2, replace=False)
+                tmp = perm[inds[0]]
+                perm[inds[0]] = perm[inds[1]]
+                perm[inds[1]] = tmp
+                idx_left.remove(inds[0])
+                idx_left.remove(inds[1])
+            return perm
+
+        ordered = ak.arange(100, dtype=dtype)
+        perm = ak.array(gen_perm(100), dtype=dtype_index)
+
+        idx_sort = ak.Series(data=ordered, index=perm).sort_index()
+        assert idx_sort.index.to_list() == ordered.to_list()
+        assert idx_sort.values.to_list() == perm.to_list()
+
+        val_sort = ak.Series(data=perm, index=ordered).sort_values()
+        assert val_sort.index.to_pandas().tolist() == perm.to_list()
+        assert val_sort.values.to_list() == ordered.to_list()
+
+    @pytest.mark.parametrize("dtype", DTYPES)
+    def test_head_tail(self, dtype):
+        size = 100
+        s = ak.Series(ak.arange(size, dtype=dtype))
+
+        head = s.head(size)
+        assert head.index.to_list() == list(range(size))
+        assert head.values.to_list() == ak.arange(size, dtype=dtype).to_list()
+
+        tail = s.tail(size)
+        assert tail.index.to_list() == ak.arange(size)[-size:size].to_list()
+        assert tail.values.to_list() == ak.arange(size, dtype=dtype)[-size:size].to_list()
```

Comment:
the `-size:size` thing doesn't really make sense to me. It's just the same as `ak.arange`
```python
In [4]: size = 10

In [5]: ak.arange(size)[-size:size]
Out[5]: array([0 1 2 3 4 5 6 7 8 9])
```

I say we just go back to what I put in the PR in ur fork

```suggestion
    def test_head_tail(self, dtype):
        n = 10
        s = ak.Series(ak.arange(n, dtype=dtype))

        for i in range(n):
            head = s.head(i)
            assert head.index.to_list() == list(range(i))
            assert head.values.to_list() == ak.arange(i, dtype=dtype).to_list()

            tail = s.tail(i)
            assert tail.index.to_list() == ak.arange(n)[-i:n].to_list()
            assert tail.values.to_list() == ak.arange(n, dtype=dtype)[-i:n].to_list()
```

Reply:
Updated